frontend/templates/product-list/server.tsx

```diff
@@ -215,6 +213,48 @@ export const getProductListServerSideProps = ({
       };
    });
 
+type GetSafeServerStateProps = {
+   appProps: AppProvidersProps;
+   indexName: string;
+   productList: ProductList;
+};
+
+async function getSafeServerState({
+   appProps,
+   indexName,
+   productList,
+}: GetSafeServerStateProps) {
+   const tryGetServerState = (productList: ProductList) => {
+      const appMarkup = (
+         <AppProviders {...appProps}>
+            <ProductListView productList={productList} indexName={indexName} />
+         </AppProviders>
+      );
+      return timeAsync('getServerState', () =>
+         getServerState(appMarkup, { renderToString })
+      );
+   };
+   try {
+      return { serverState: await tryGetServerState(productList) };
+   } catch (e) {
+      console.error('Error getting instantsearch server state', e);
+      Sentry.withScope((scope) => {
+         scope.setExtra('Productlist', productList);
+         scope.setExtra('Filters from Strapi', productList?.filters);
+         Sentry.captureException(e);
+      });
```

Comment:
Oops, I forgot to drop the code that reports to Sentry now that we're rendering a notice on the ui. I'll open another pull.

Reply:
https://github.com/iFixit/react-commerce/pull/1571